Cython/Compiler/PyrexTypes.py

```diff
@@ -3132,8 +3132,12 @@ def declaration_code(self, entity_code,
         if (pyrex or for_display) and not self.return_type.is_pyobject:
             if self.exception_value and self.exception_check:
                 trailer = " except? %s" % self.exception_value
-            elif self.exception_value:
+            elif self.exception_value and not self.exception_check:
                 trailer = " except %s" % self.exception_value
+            elif not self.exception_value and self.exception_check:
+                trailer = " except +"
```

Comment:
Apologies for not making this clearer, but taking over these two lines was equivalent to mapping `exc_val == '+'` to `exc_check=False, exc_val=True`. So unless/until you do that, you'll need to remove these two lines again.